### ISSUE TITLE
Pyrosium and Cryostylane react with any amount of oxygen instantly

### ISSUE THREAD

**Issue by Vidiveni** [tgstation/tgstation#38887]
[Round ID] etc: Don't remember. Any. Terry. 

**Reproduction:** 
1. Put 1u of pyrosium or cryostylane in a large beaker. 
2. Put the beaker in a chem dispenser. 
3. Dispense 50u oxygen. 

Result: Instant reaction x50 consuming all oxygen. 
Expected result: Slow reaction consuming 1u oxygen every second or so.  Like in the description. 

Also I'm not sure if I read the code right, but seems the two chems heat/cool 10 Kelvin per oxygen in beakers, but 15-30 Kelvin per oxygen in carbon bodies. Inconsistent but I could be wrong. 

https://github.com/tgstation/tgstation/blob/master/code/modules/reagents/chemistry/recipes/pyrotechnics.dm#L360
vs
https://github.com/tgstation/tgstation/blob/master/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm#L209

**Chem description:**
"Comes into existence at 20K. As long as there is sufficient oxygen for it to react with, Pyrosium **slowly** heats all other reagents in the container."

**Comment by 81Denton** [comment]
I was gonna blame this on the wiki until I read the CL. Looks like it's intended to be delayed:

```
rscadd: "Adds Pyrosium. This reagent heats up mobs by 30 degrees every 3 seconds if it has oxygen to react with. Useful for when you want a delayed mix inside of someone."
rscadd: "Adds Cryostylane.  This reagent does the opposite of Pyrosium, but it still requires oxygen to react with."
```

**Comment by jegub** [comment]
They were originally a progressive series of reactions, I think it was changed to instant because keeping track was inefficient. After the heating/cooling amount was reduced (to prevent exploding meth foam and the concomitant explosion notification spam to admins?) they didn't have much in the way of uses that can't be replicated with a custom timer grenade.

**Comment by Vidiveni** [comment]
I mean those delayed reactions would be pretty fun to play with. Sounds like there is a design decision to make here.

**Comment by jegub** [comment]
My knowledge of the matter is limited to a late-night OOC exchange that I had to leave early, from what I vaguely recall these two chemicals required all chemical containers to check if this unique behaviour was to be triggered, either that or they required unique variables to be added for all chemical reagents.
If the same effect could be achieved more simply then it would be nice to have back, if only because you could use cryostylane for purely chemical cryo healing in a pinch.

**Comment by vuonojenmustaturska** [comment]
@81Denton I changed this in #31281

tldr the way it used to work was stupid and the reagents were not used for the delay action, in fact only pyrosium was used at all and only as a trigger to lift reagent temperature one degree across a threshold. so I saw no reason to spend time making the perfect fix when pyro and cryo immediately became more useful with the removal of the delay.

**Comment by Vidiveni** [comment]
If it's intended then this issue could be closed as soon as the in-game reagent description changes to reflect how it actually works.

**Comment by Vidiveni** [comment]
Meh. Probably nobody reads in-game descriptions anyway and I've changed the wiki.
